### ISSUE TITLE
Faded icons of minimized windows: so light they're hard to see

### ISSUE THREAD

**Issue by thomas2net** [mate-desktop/mate-panel#435]
Dear mate-panel developers,

I use Linux Mint LMDE with Mate. I'm having difficulties recognizing the icons of minimized windows in the Mate Panel. The reason is that I have several windows whose icon is mostly white or a light color. Once these are minimized and their icon made lighter by Mate Panel it becomes difficult differentiating the tabs by looking at the icon.
Is there a way to cancel or customize the lighter appearance of the icons?
Which source code file can I look into to fix this?
Could you consider an alternative, e.g., drawing a small down-pointing triangle that partly overlaps onto the icon?

Many thanks!

**Comment by raveit65** [comment]
Which theme are you using?
The only way to fix that is to change the bg color of the so called button of wncklist in theme.

**Comment by thomas2net** [comment]
Hi raveit65,
I use the default Mint-X.
After searching, greping and RTFMing a bit I have not found what controls the lighting (or fading? -- I see several "fade" occurrences but I'm not sure they apply to the minimized icon). This possibly happens around:
/usr/share/themes/Mint-X/gtk-2.0/gtkrc
/usr/share/themes/Mint-X/gtk-3.0/gtk-widgets.css

Thanks for your insight!

**Comment by raveit65** [comment]
Well. Mint-X theme is from linuxmint itself and not from MATE.
So i suggest to open a report there.
This is the responsible file /usr/share/themes/Mint-X/gtk-2.0/gtkrc for a gtk2 desktop.
@JosephM
any thoughts?

**Comment by martinvonwittich** [comment]
The icon dimming is AFAIK done by libwnck. Years ago, when I was still using GNOME2, I had maintained an Ubuntu PPA where I had commented out the code line that was responsible: https://launchpad.net/~martin.von.wittich/+archive/ubuntu/libwnck-nodimming

There were several bug reports about this, for example here, but unfortunately this was never fixed upstream: https://bugzilla.gnome.org/show_bug.cgi?id=552416

As far as I can tell, MATE used to have a libmatewnck fork, but that seems to be discontinued now ( https://github.com/mate-desktop-legacy-archive/libmatewnck ). On my Ubuntu system it looks like the window list is using the GNOME libwnck:

```
martin@martin ~ % ps aux|grep wnck-applet
martin    2853  0.0  0.1 539448 29396 ?        Sl   16:23   0:03 /usr/lib/mate-panel/wnck-applet
martin    9031  0.0  0.0  15796   980 pts/3    S+   19:32   0:00 grep --colour=auto wnck-applet

martin@martin ~ % ldd /usr/lib/mate-panel/wnck-applet | grep -i wnck
        libwnck-1.so.22 => /usr/lib/x86_64-linux-gnu/libwnck-1.so.22 (0x00007fc115920000)

martin@martin ~ % dpkg -S /usr/lib/x86_64-linux-gnu/libwnck-1.so.22 
libwnck22:amd64: /usr/lib/x86_64-linux-gnu/libwnck-1.so.22

martin@martin ~ % aptitude show libwnck22
Package: libwnck22                       
State: installed
Automatically installed: no
Multi-Arch: same
Version: 1:2.30.7-5ubuntu1.1
Priority: optional
Section: universe/libs
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Uncompressed Size: 338 k
Depends: libatk1.0-0 (>= 1.12.4), libc6 (>= 2.14), libcairo2 (>= 1.2.4), libgdk-pixbuf2.0-0 (>= 2.22.0), libglib2.0-0 (>= 2.37.3),
         libgtk2.0-0 (>= 2.22.0), libpango-1.0-0 (>= 1.14.0), libstartup-notification0 (>= 0.4), libx11-6, libxres1, libwnck-common
Conflicts: libwnck-1
Breaks: libwnck22:i386 (!= 1:2.30.7-5ubuntu1.1)
Replaces: libwnck-1, libwnck22:i386 (< 1:2.30.7-5ubuntu1.1)
Description: Window Navigator Construction Kit - runtime files
 A library to use for writing pagers and task lists. This variant of libwnck provides WNCK for GTK-2 based applications. 
 
 This package contains runtime files.
Homepage: https://developer.gnome.org/libwnck/2.30/
```

**Comment by lukefromdc** [comment]
I've never had this issue once in GTK 3 builds. For mate-panel 1.17, which is GTK3 only, this line appears in configure.ac:
LIBWNCK_REQUIRED=3.0.0
My guess is libwnck 3.0 does not do this or some change in how it is used prevents it, so this issue goes away with the GTK3 migration,

**Comment by thomas2net** [comment]
Yes. Icons in panel, once minimized, still appear faded and I still find them not that easy to recognize. 
7 years later, I'm not getting any younger!
Thanks to anyone finding a fix to this. Don't close this issue unless it's fixed.

**Comment by thomas2net** [comment]
![Screenshot at 2023-07-21 16-50-19](https://github.com/mate-desktop/mate-panel/assets/16047553/63e24044-fe2a-45e3-91a4-0328d6cafdde)

The faded Firefox icons are reasonably recognizable.
The faded Chromium icon could be confused with others.
Challenge: Can you recognize the icon at the bottom of the list?

**Comment by raveit65** [comment]
I don't think that our small team can solve you wish.
Maybe you can hire a external developer?

**Comment by thomas2net** [comment]
If someone can point me to the line that's responsible for fading out the icons, I'd be happy to comment out that line without hiring an external developer.

**Comment by cwendling** [comment]
@thomas2net It's probably somewhere in `mate-panel` or `libwnck`.  Closer I can't tell without diving into this myself :)
